packages/core/rpc/ElectronRpcDriver.ts

```diff
@@ -19,38 +19,63 @@ async function wait(ms: number): Promise<void> {
   })
 }
 
+type BackendConfig = { workerCreationChannel: string }
+type WorkerBootConfig = { plugins: PluginDefinition[] }
+
 class WindowWorkerHandle {
   private ipcRenderer: import('electron-better-ipc-extra').RendererProcessIpc
 
   private window: import('electron').BrowserWindow
 
+  private config: WorkerBootConfig
+
   private ready = false
 
   constructor(
     ipcRenderer: import('electron-better-ipc-extra').RendererProcessIpc,
     window: import('electron').BrowserWindow,
+    config: WorkerBootConfig,
   ) {
     this.ipcRenderer = ipcRenderer
     this.window = window
+    this.config = config
   }
 
   destroy(): void {
     this.window.destroy()
   }
 
+  // waits for a new worker to start, and then sends it its bootstrap configuration
+  async setup() {
+    if (!this.ready) {
+      let readyForConfig = false
+      while (!readyForConfig) {
+        await wait(1000)
+        readyForConfig = !!(await this.ipcRenderer.callRenderer(
+          this.window,
+          'ready_for_configuration',
+        ))
+      }
+
+      const result = await this.ipcRenderer.callRenderer(
+        this.window,
+        'configure',
+        this.config,
+      )
+      if (!result) {
+        throw new Error('failed to configure worker')
+      }
+    }
+    this.ready = true
```

Comment:
perhaps move this line into the conditional above it so it isn't set again to true every time setup is called